dark/process.py

```diff
@@ -31,8 +39,14 @@ def dryRun(self):
         return self._dryRun
 
     def execute(
-        self, command, dryRun=None, useStderr=True, stdout=False, stderr=False, **kwargs
-    ):
+        self,
+        command: Union[str, list[str]],
+        dryRun: Optional[bool] = None,
+        useStderr: bool = True,
+        stdout: Optional[Union[bool, TextIO, StringIO]] = False,
+        stderr: Optional[Union[bool, TextIO, StringIO]] = False,
```

Comment:
I would remove the `bool` and change the default to `None`, it seems unnecessary to have two false-y values meaning the same thing.

Reply:
Thanks, I agree it's a bit weird.

But they don't mean the same thing. If no value is given, the default (`False`) will result in the value from `self` being used. If `None` is passed, there will be no output.

Why? I want the user to be able to pick between three options for any given call to `execute`: 1) the default behaviour (originally passed to `__init__`), 2) no output (by explicitly passing `None`), or 3) to a specific file descriptor.  Option 2 is incompatible with using `None` as the default value.